mesonbuild/dependencies/base.py

```diff
@@ -1479,8 +1481,25 @@ def _detect_dep(self, name: str, modules: T.List[T.Tuple[str, bool]], components
                 for j in otherDeps:
                     if j in self.traceparser.targets:
                         targets += [j]
-                    elif reg_is_lib.match(j) or os.path.exists(j):
+                    elif reg_is_lib.match(j):
                         libraries += [j]
+                    elif os.path.isabs(j) and os.path.exists(j):
+                        libraries += [j]
+                    elif mesonlib.is_windows() and reg_is_maybe_bare_lib.match(j):
+                        # On Windows, CMake library dependencies can be passed as bare library names,
+                        # e.g. 'version' should translate into 'version.lib' or 'version.dll' depending
+                        # on the link mode. CMake brute-forces a combination of prefix/suffix
+                        # combinations to find the right library, however as we do not have a compiler
+                        # environment available to us, we cannot do the same, but must assume any
+                        # bare argument passed which is not also a CMake target must be a library we
+                        # should try to link against
+                        if self.static:
+                            libraries += ["{}.lib".format(j)]
+                        else:
+                            libraries += ["{}.dll".format(j)]
```

Comment:
I wasn't aware that you could link directly to .dlls… I'm pretty sure you have to link to a .lib always, it's just that a static .lib contains code, where a dynamic .lib only contains stubs which add entries to the import tables.

Reply:
Oh, OK. So should we always just pass `.lib` in this case?

Reply:
Yeah, I think so. I think I've seen tools that can generate a corresponding import lib from from a dll, but the linker doesn't take dlls directly as inputs.